packages/toolkit/src/createSlice.ts

```diff
@@ -585,17 +849,167 @@ function getType(slice: string, actionKey: string): string {
   return `${slice}/${actionKey}`
 }
 
-interface BuildCreateSliceConfig {
-  creators?: {
-    asyncThunk?: typeof asyncThunkCreator
+export const reducerCreator: ReducerCreator<ReducerType.reducer> = {
+  type: ReducerType.reducer,
+  create(caseReducer: CaseReducer<any, any>) {
+    return Object.assign(
+      {
+        // hack so the wrapping function has the same name as the original
+        // we need to create a wrapper so the `reducerDefinitionType` is not assigned to the original
```

Comment:
we need to assign the _reducerDefinitionType brand to the reducer we return, so it can be passed to the right handler. however, we probably don't want to assign it to the function reference passed to the creator. this creates a new function reference with the same name as the one passed in (for trace purposes) and assigns the brand to that instead